File: converted_source/unoted.pas.java
// unit unoted

from conversion_common import *;
import updform;
import delphi_compatability;

// const
kBetweenGap = 4;



class TNoteEditForm extends PdForm {
    public TButton OK;
    public TButton cancel;
    public TMemo noteEditMemo;
    public TCheckBox wrapLines;
    public TSpeedButton helpButton;
    
    //$R *.DFM
    public void FormCreate(TObject Sender) {
        this.noteEditMemo.WordWrap = UNRESOLVED.domain.options.noteEditorWrapLines;
        this.wrapLines.Checked = UNRESOLVED.domain.options.noteEditorWrapLines;
    }
    
    public void FormResize(TObject Sender) {
        this.OK.Left = this.ClientWidth - this.OK.Width - kBetweenGap;
        this.cancel.Left = this.OK.Left;
        this.helpButton.Left = this.OK.Left;
        this.wrapLines.Left = this.OK.Left + kBetweenGap;
        this.noteEditMemo.SetBounds(kBetweenGap, kBetweenGap, this.OK.Left - kBetweenGap * 2, this.ClientHeight - kBetweenGap * 2);
    }
    
    public void wrapLinesClick(TObject Sender) {
        UNRESOLVED.domain.options.noteEditorWrapLines = this.wrapLines.Checked;
        this.noteEditMemo.WordWrap = UNRESOLVED.domain.options.noteEditorWrapLines;
    }
    
    public void helpButtonClick(TObject Sender) {
        delphi_compatability.Application.HelpJump("Editing_plant_notes");
    }
    
}

File: converted_source/unoted.pas.py
# unit unoted

from conversion_common import *
import udomain
import updform
import delphi_compatability

# const
kBetweenGap = 4

class TNoteEditForm(PdForm):
    def __init__(self):
        self.OK = TButton()
        self.cancel = TButton()
        self.noteEditMemo = TMemo()
        self.wrapLines = TCheckBox()
        self.helpButton = TSpeedButton()
    
    #$R *.DFM
    def FormCreate(self, Sender):
        self.noteEditMemo.WordWrap = udomain.domain.options.noteEditorWrapLines
        self.wrapLines.Checked = udomain.domain.options.noteEditorWrapLines
    
    def FormResize(self, Sender):
        self.OK.Left = self.ClientWidth - self.OK.Width - kBetweenGap
        self.cancel.Left = self.OK.Left
        self.helpButton.Left = self.OK.Left
        self.wrapLines.Left = self.OK.Left + kBetweenGap
        self.noteEditMemo.SetBounds(kBetweenGap, kBetweenGap, self.OK.Left - kBetweenGap * 2, self.ClientHeight - kBetweenGap * 2)
    
    def wrapLinesClick(self, Sender):
        udomain.domain.options.noteEditorWrapLines = self.wrapLines.Checked
        self.noteEditMemo.WordWrap = udomain.domain.options.noteEditorWrapLines
    
    def helpButtonClick(self, Sender):
        delphi_compatability.Application.HelpJump("Editing_plant_notes")
    
